samples/apply-dictionary-renderer-to-graphics-overlay/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertographicsoverlay/components/ApplyDictionaryRendererToGraphicsOverlayViewModel.kt

```diff
@@ -0,0 +1,252 @@
+import java.io.File
+
+/**
+ * ViewModel for the "Apply dictionary renderer to graphics overlay" sample.
+ */
+class ApplyDictionaryRendererToGraphicsOverlayViewModel(private val app: Application) : AndroidViewModel(app) {
+
+    // The scene shown in the SceneView composable
+    val arcGISScene = ArcGISScene(basemapStyle = BasemapStyle.ArcGISTopographic)
+
+    // Graphics overlay that will hold the message graphics
+    val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    // SceneViewProxy to enable programmatic viewpoint changes
+    val sceneViewProxy = SceneViewProxy()
+
+    // Used to display error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Provision path where downloaded sample assets will be placed by the downloader activity
+    private val provisionPath: String by lazy {
+        app.getExternalFilesDir(null)?.path.toString() + File.separator + app.getString(R.string.apply_dictionary_renderer_to_graphics_overlay_app_name)
+    }
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
+            // Load the scene first
+            arcGISScene.load().onFailure { throwable ->
+                messageDialogVM.showMessageDialog(throwable)
+            }
+
+            // Create and apply dictionary renderer from a web style
+            val dictionaryRendererDeferred = viewModelScope.async(Dispatchers.IO) {
+                createMil2525dDictionaryRenderer()
+            }
+
+            // Create the point graphics in separate coroutine from a local XML file
+            val graphicsDeferred = viewModelScope.async(Dispatchers.IO) {
+                makeMessageGraphics()
+            }
+
+            val dictionaryRenderer = dictionaryRendererDeferred.await()
+            val pointGraphics = graphicsDeferred.await()
+
+            // sets the graphics in the overlay
+            graphicsOverlay.apply {
+                renderer = dictionaryRenderer
+                graphics.addAll(pointGraphics)
+            }
+
+            // Sets the camera to look a the graphics in the graphics overlay
+            val extent = graphicsOverlay.extent ?: run {
+                messageDialogVM.showMessageDialog(
+                    title = "Error",
+                    description = "The extent of the graphics overlay is null."
+                )
+                return@launch
+            }
+
+            // Set the viewpoint of the scene view using a camera
+            val camera = Camera(
+                lookAtPoint = extent.center,
+                distance = 15_000.toDouble(),
+                heading = 0.toDouble(),
+                pitch = 70.toDouble(),
+                roll = 0.toDouble()
+            )
+            sceneViewProxy.setViewpointCamera(camera)
+        }
+    }
+
+    /**
+     * Create and load a DictionarySymbolStyle from a web style and use it to create a DictionaryRenderer.
+     */
+    private suspend fun createMil2525dDictionaryRenderer(): DictionaryRenderer? {
```

Comment:
If using the above suggestions: 
```suggestion
    private suspend fun createMil2525dDictionaryRenderer(): Result<DictionaryRenderer> {
```